lib/python/rose/config_processors/fileinstall.py

```diff
@@ -250,6 +250,19 @@ def _process(self, conf_tree, nodes, loc_dao, **kwargs):
                         if prev_path != path:
                             target.is_out_of_date = True
                             break
+                # See if any sources have changed names.
+                if not target.is_out_of_date:
+                    conn = loc_dao.get_conn()
+                    prev_dep_locs = conn.execute(
+                        "SELECT * FROM dep_names WHERE name=?", [target.name]
+                    ).fetchall()
+                    prev_dep_locs = [i[1] for i in prev_dep_locs]
+                    prev_dep_locs = [loc_dao.select(i) for i in prev_dep_locs]
+                    if (
+                        [i.name for i in prev_dep_locs] !=
+                        [i.name for i in target.dep_locs]
```

Comment:
This ensures strict order is preserved, which may or may not be a good thing, will have to do some digging, my questions:

* Are `prev_dep_locs` always going to be stored in the same order as `target.dep_locs`?
  * I.E. Even if you don't change the `source` list could this equality resolve to `False` anyway.
* Does the order matter?

Reply:
Answer, yes the order does matter:

```
[file:multiple]
source=a b c
```

Would result in a file that looks like this:

```
cat a
cat b
cat c
```

Reply:
I meant to sort, and forgot. Oops.

Reply:
No oops at all, I think the order can matter.

Reply:
After a bit of a play I'm happy that dep_names are stored in the right order too.

Reply:
```suggestion
                        sorted([i.name for i in prev_dep_locs]) !=
                        sorted([i.name for i in target.dep_locs])
```

Reply:
Agh, no, my comments here were backing up your previous decision *not* to sort.

The order of sources can effect the resulting output so exact (unsorted) equality must be tested.